[fix/various] Various fixes

felix-schwarz opened the pull request:
## Description
This PR aims to fix various minor issues in the app and file provider:
- [x] extend hardcoded MIMEType/suffix -> UTI mappings to more common file types (prevents "disabled" files in File Provider)
- [x] thumbnails mixed up after removal of files (as reported by @TheOneRing)

## Related Issue

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

felix-schwarz committed:
- FileProvider: extend hardcoded MIME/Suffix -> UTI maps so files of these types are properly displayed in the Files app

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/ios-app?pullRequest=713) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/ios-app?pullRequest=713) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: mneuwert<br/>:white_check_mark: hosy<br/>:x: felix-schwarz<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/ios-app?pullRequest=713) it.</sub>

felix-schwarz committed:
- add new ItemContainer protocol allowing the retrieval of an item from another object
- UIImageView+Thumbnails
	- fix item version comparison for thumbnails requested from the SDK (using ItemContainer)
	- add item version comparison for thumbnails generated locally via QLThumbnailGenerator (using ItemContainer)

jesmrec commented:
> extend hardcoded MIMEType/suffix -> UTI mappings to more common file types (prevents "disabled" files in File Provider)

The list of file extensions mentioned in `ownCloud File Provider/OCItem+FileProviderItem.m` is currently supported in Files App, so the files are not greyed out anymore.

> thumbnails mixed up after removal of files (as reported by @TheOneRing)

I've seen that problem in the past, but, not able to reproduce in the current `master`. So, the feedback of @TheOneRing will be helpful here. In the current branch is not reproducible either

TheOneRing commented:
I performed some quick tests on 11.4 159 and it seemed to be fixed

jesmrec commented:
Thanks @TheOneRing this is the last evidence needed to move forward.

felix-schwarz committed:
- Update SDK
